### PR TITLE
mailpile: 0.4.1 -> 0.5.2

### PR DESCRIPTION
**Pull request by spinus**
###### Motivation for this change

New version of mailpile is out, current version is waaaay to old.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by spinus** [comment]
I a problem with upgrading this package, I can't build it properly. There is no issues when I run "python setup.py bdist_wheel" when I run this manually outside of nixpkgs tree. Maybe someone knows what the issue could be and what is potential solution?

File: pkgs/applications/networking/mailreaders/mailpile/default.nix [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 python2Packages.buildPythonApplication rec {
   name = "mailpile-${version}";
-  version = "0.4.1";
+  version = "0.5.2";
 
   src = fetchgit {
```
Comment (Mic92) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]: this should be upgraded to `fetchFromGithub`.

Comment (fpletz) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]: The repo uses git submodules which aren't included in the Github tarball.

**Comment by Mic92** [comment]
Build is fixed now, but the released pre-release is still to old. At least gpg integration did not work for me. Probably a commit which include this https://github.com/mailpile/Mailpile/pull/1621 is required.

**Comment by spinus** [comment]
@Mic92 thank you for fixing it, I'll try to work on it.

File: pkgs/applications/networking/mailreaders/mailpile/default.nix [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-    url = "git://github.com/pagekite/Mailpile";
+    url = "https://github.com/mailpile/Mailpile";
     rev = "refs/tags/${version}";
-    sha256 = "118b5zwfwmzj38p0mkj3r1s09jxg8x38y0a42b21imzpmli5vpb5";
+    sha256 = "1d2b776x9134sv67pylfkvf1dd4vs5pvgrngpmshrsjgsib13dx5";
   };
 
   patchPhase = ''
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]: I think its better to have `postPatch` here. That way one can still pass in patches.

Comment (spinus) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]: done

File: pkgs/applications/networking/mailreaders/mailpile/default.nix (hunk 1 above)
Comment (spinus) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]: hmm, I moved to fetchFromGitHub but it still building, I'll try to verify later if it's working

**Comment by spinus** [comment]
it looks like this packages was not released for a long time, I just used master for now. Any ideas if we should keep latest officially released one or stick with master? (maybe we should have 0.5.2 and master as unstable package)

**Comment by FRidh** [comment]
> (maybe we should have 0.5.2 and master as unstable package)

Yes.

**Comment by Mic92** [comment]
@spinus if this is required to get gpg working, I am ok with that. Also mailpile is still in early development.

**Comment by spinus** [comment]
I wrote to mailpile developers to ask about little clarification on that, here is what I've got:

> If you can, please remove 0.5.2 from NixOS. It is very buggy and
> has serious known security vulnerabilities.
> 
> There will be a release candidate for 1.0 sometime this year;
> until then I am asking the community to please NOT package
> Mailpile at all, as packages will give a false expectation of
> stability, will attract the wrong kind of user and will
> ultimately disappoint people.
> 

So the question is, should we remove mailpile from nixos as it's not ready yet, or we want to keep it anyway (in this case I assume we should clearly communicate that this software is not ready to be used "in production"). What do you think people?
> I know that's not what people want to hear, but that's the
> current state of things.

**Comment by FRidh** [comment]
Simple, remove it, the maintainer is very clear about it...

**Comment by BjarniRunar** [comment]
Thanks @FRidh and @spinus for reaching out! Please feel free to be in touch at any time if you have questions, this should only be a temporary situation.

**Comment by grahamc** [comment]
Can we get a list of these known vulnerabilities, then:

1. set `meta.knownVulnerabilities` on 17.03, unstable
2. mark it as broken on 16.09

**Comment by BjarniRunar** [comment]
I don't have a curated list of vulnerabilities; I have a bug tracker full of open and closed issues, some of which almost certainly have a security impact.

**Comment by BjarniRunar** [comment]
Basically, you've now demonstrated *exactly* why I'm asking this not be packaged. We don't have the stability, processes or resources to support that. It should never have been packaged in the first place. The only "consolation" is that those old releases were so buggy that I'd be amazed if anyone ever managed to use them for anything of import.

**Comment by grahamc** [comment]
I've marked it as broken in our stable distribution: 8605d317d1

**Comment by grahamc** [comment]
Removing it is a bit tricky due to the service, but would happily merge a PR doing that.

I marked it as insecure on next-stable (17.03, 85b47bbd5e) and unstable (6111f6a756) just in case the removal doesn't happen.

File: pkgs/applications/networking/mailreaders/mailpile/default.nix [hunk 3]
```diff
@@ -1,21 +1,27 @@
-{ stdenv, fetchgit, python2Packages, gnupg1orig, makeWrapper, openssl }:
+{ stdenv, fetchFromGitHub, python2Packages, gnupg1orig, openssl, git }:
 
 python2Packages.buildPythonApplication rec {
   name = "mailpile-${version}";
-  version = "0.4.1";
+  version = "0.5.x-git-master-20170301";
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/mailpile/default.nix]:
```
pname = "mailpile";
version = "unstable-2017-03-01";
name = "${pname}-${version}";
```

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
I've mentioned one more improvement in case one do wants to get it in. Merging this upgrade I don't mind now that it is marked as insecure.

**Comment by spinus** [comment]
ok, as mailpile is marked as insecure and there is advice not to releases for anything below 1.0 I assume we can close that

**Pull request closed without merging**